nbcelltests/test.py

```diff
@@ -13,120 +13,29 @@
-            skiptest = "@nbcelltests.tests_vendored.unittest.skip('no test supplied')\n" + INDENT
-        elif not cell_injected_into_test(test_lines):
-            skiptest = "@nbcelltests.tests_vendored.unittest.skip('cell code not injected into test')\n" + INDENT
-        else:
-            skiptest = ""
-
-        # TODO: Use namedtuples for these:
-        cells.append([code_cell, [], "%sdef test_code_cell_%d(self):\n" % (skiptest, code_cell)])
-
-        if skiptest:
-            cells[-1][1].append(INDENT + 'pass # code cell %d was skipped\n' % code_cell)
-            continue
-
-        for test_line in test_lines:
-            if test_line.strip().startswith(r"%cell"):
-                # add comment in test for readability
-                cells[-1][1].append(INDENT + test_line.replace(r'%cell', '# Cell {' + str(code_cell) + '} content\n'))
-
-                # add all code for cell
-                for cell_line in cell['source'].split('\n'):
-                    # TODO: is this going to replace %cell appearing in a comment?
-                    cells[-1][1].append(INDENT + test_line.replace('\n', '').replace(r'%cell', '') + cell_line + '\n')
-
-            # else just write test
-            else:
-                cells[-1][1].append(INDENT + test_line)
-                if not test_line[-1] == '\n':
-                    cells[-1][1][-1] += '\n'
-    return cells
-
-
-def writeout_test(fp, cells, kernel_name):
-    fp.write(BASE.format(kernel_name=kernel_name))
-
-    # write out source of cells+tests
-    fp.write(INDENT + "cells_and_tests = {")
-    for i, code, _ in cells:
-        fp.write('\n')
-        fp.write(INDENT * 2 + '%d: """\n' % i)
-        for line in code:
-            fp.write(INDENT + line)
-        fp.write(INDENT * 2 + '""",\n')
-    fp.write(INDENT + "}\n")
-
-    # write out test methods
-    for i, _, meth in cells:
-        fp.write('\n')
-        fp.write(INDENT + meth)
-        fp.write(INDENT * 2 + 'self.run_test(%d)\n' % i)
-
-
-def writeout_cell_coverage(fp, cell_coverage, metadata):
-    if cell_coverage:
-        fp.write(INDENT + 'def test_cell_coverage(self):\n')
-        fp.write(
-            2 *
-            INDENT +
-            'assert {cells_covered} >= {limit}, "Actual cell coverage {cells_covered} < minimum required of {limit}"\n\n'.format(
-                limit=cell_coverage,
-                cells_covered=get_coverage(metadata)))
-
-
-def run(notebook, rules=None, filename=None):
+
+def run(notebook, rules=None, filename=None, override_kernel_name=""):
     """Runs no tests: just generates test script for supplied notebook."""
     nb = nbformat.read(notebook, 4)
-    name = filename or notebook[:-6] + '_test.py'  # remove .ipynb, replace with _test.py
-
-    kernel_name = nb.metadata.get('kernelspec', {}).get('name', 'python')
-    cells = assemble_code(nb)
+    name = filename or notebook[:-6] + '_test.py'
```

Comment:
`os.path.splitext(notebook)[0]` ?

Reply:
I agree - not changed as part of this PR though :)

Reply:
Oh, I removed a comment. Ha ha, yeah I should fix this line then since I touched it.